rrule/src/core/rruleset.rs

```diff
@@ -26,6 +37,13 @@ impl Default for RRuleSet {
 }
 
 impl RRuleSet {
+    pub fn new(dt_start: DateTime) -> Self {
+        Self {
+            dt_start,
+            ..Default::default()
+        }
+    }
+
     pub fn rrule(&mut self, rrule: RRule) {
```

Comment:
You are right, nice catch, fields shouldn't be public.
But I think method are correct atm. You can change `Unvalidated` one, but there is no method on the `Validated`. So it's immutable!

Reply:
The hard part is that I have to update all the tests :))
But np, it will take more time :P

Reply:
> You can change Unvalidated one, but there is no method on the Validated.

Can you elaborate on why this means it is better to make the fields private rather than changing the method signature to take `RRule<Unvalidated>`.

A benefit of accepting an unvalidated `RRule` is that the user don't have to call the `RRule::validate` method, to call `RRule::validate` they also need the start date which is a field on `RRuleSet` so it makes sense that the validation happens inside `RRuleSet`.

Changing all tests sound like a daunting task so let's be sure about the change before actually doing it :D

Reply:
I think you mixed two comments, or I misunderstood the first comment in this thread. There is [another comment](https://github.com/fmeringdal/rust-rrule/pull/60#discussion_r878353081) for `RRuleSet` and this one is for `RRule` I think!

Anyway, I agree to make both `RRule` and `RRuleSet` properties, private (or maybe `pub(crate)`).

Reply:
> Can you elaborate on why this means it is better to make the fields private rather than changing the method signature to take RRule<Unvalidated>?

And here, I meant, when `RRule` fields become private, you cannot make a `Validated` invalid `RRule`. So it will be safe.

Reply:
I [made](https://github.com/fmeringdal/rust-rrule/pull/60/commits/22db507391c2e7c31a35a5db000987f44da4372c) all `RRule` and `RRuleSet` fields private (pub(crate)).
Now the code you pasted here shouldn't work and the API should be safe, yes?

Reply:
And to fix the `could not compile rrule due to 2771 previous errors` issue, I moved the tests inside.